darwin/pytorch/utils.py

```diff
@@ -0,0 +1,288 @@
+        remote_datasets = [dataset.slug for dataset in client.list_remote_datasets()]
+        if dataset_name in remote_datasets:
+            dataset = client.get_remote_dataset(slug=dataset_name)
+            progress, _count = dataset.pull(image_status=image_status)
+            with tqdm(total=_count, desc=f"Downloading '{dataset_name}' dataset") as pbar:
+                for _ in progress():
+                    pbar.update()
+        else:
+            raise ValueError(f"Could not find dataset {dataset_name} in Darwin.")
+
+    # Find annotations and create folders
+    root = Path(client.project_dir) / dataset_name
+    annot_path = root / "annotations"
+    annot_files = [f for f in annot_path.glob('*.json')]
+    num_images = len(annot_files)
+    lists_path = root / "lists"
+    mkdirs(lists_path)
+
+    # Extract classes
+    fname = lists_path / "classes.txt"
+    if not fname.exists():
+        # Extract list of classes
+        classes, idx_to_classes = extract_classes(annot_files)
+        classes_names = [k for k in classes.keys()]
+        classes_names.insert(0, '__background__')
+        with open(lists_path / 'classes.txt', 'w') as f:
+            for c in classes_names:
+                f.write(f"{c}\n")
+
+    # Create split
+    split_id = f"split_val{val_percentage}_test{test_percentage}"
+    if split_seed is not None:
+        np.random.seed(split_seed)
+        split_id += f"_seed{split_seed}"
+
+    split_path = lists_path / split_id
+    if not split_path.exists() or force_resplit:
+        mkdirs(split_path)
+        num_train = int(num_images * (1 - (val_percentage + test_percentage)))
+        num_test = int(num_images * test_percentage)
+        num_val = num_images - num_train - num_test
+
+        indices = np.random.permutation(num_images)
+        train_idx = indices[:num_train]
+        val_idx = indices[num_train:num_train+num_val]
+        test_idx = indices[num_train+num_val:]
+
+        # Write files
+        with open(split_path / 'train.txt', 'w') as f:
+            for i in train_idx:
+                f.write(f"{annot_files[i].stem}\n")
+        if num_val > 0:
+            with open(split_path / 'val.txt', 'w') as f:
+                for i in val_idx:
+                    f.write(f"{annot_files[i].stem}\n")
+        if num_test > 0:
+            with open(split_path / 'test.txt', 'w') as f:
+                for i in test_idx:
+                    f.write(f"{annot_files[i].stem}\n")
+
+    return root, split_id
+
+
+# VISUALIZATION
+def visualize_mask_output(image, masks, classes, colors, threshold=0.5):
+    W, H = image.size
+    outmask = np.zeros((H, W, 4), dtype=np.uint8)
+    for mask, c in zip(masks, classes):
+        color = np.append(colors[c], 128)
+        outmask[mask > threshold] = color
+
+    outmask = Image.fromarray(outmask)
+    image.paste(outmask, (0, 0), outmask)
+    return image
+
+
+def get_colors(n, name='hsv'):
```

Comment:
Are we using this function anywhere?

Reply:
It's another visualization-related function. I'll remove it as well